runtime/services/catalog/artifacts/artifacts.go

```diff
@@ -70,6 +71,18 @@ func Write(ctx context.Context, repoStore drivers.RepoStore, instID string, cata
 
 var regex = regexp.MustCompile("^[a-zA-Z_][a-zA-Z0-9_]*$")
 
-func isValidName(itemName string) bool {
+func IsValidName(itemName string) bool {
 	return regex.MatchString(itemName)
 }
+
+var invalidChars = regexp.MustCompile(`[^a-zA-Z_\d]`)
+
+// SanitizedName returns a sanitized name for an artifact from file path.
+func SanitizedName(filePath string) string {
+	name := invalidChars.ReplaceAllString(fileutil.Stem(filePath), "_")
```

Comment:
Should we also use sanitized name for file uploads? Also, should we check that artifact names are sanitized in reconcile, and return an error if an artifact has a non-sanitized name?

Reply:
For file upload we need a good duplicate name check before we can support it. Working on it in another branch.

For validation during reconcile, it already happens during an artifact read.